src/main/java/com/sprint/mission/discodeit/service/basic/BasicChannelService.java

```diff
@@ -0,0 +1,121 @@
+package com.sprint.mission.discodeit.service.basic;
+
+import com.sprint.mission.discodeit.DTO.Channel.CreatePrivateChannelDTO;
+import com.sprint.mission.discodeit.DTO.Channel.CreatePublicChannelDTO;
+import com.sprint.mission.discodeit.DTO.Channel.FindChannelDTO;
+import com.sprint.mission.discodeit.DTO.Channel.UpdateChannelDTO;
+import com.sprint.mission.discodeit.DTO.ReadStatus.CreateReadStatusDTO;
+import com.sprint.mission.discodeit.Enum.ChannelType;
+import com.sprint.mission.discodeit.entity.Channel;
+
+import com.sprint.mission.discodeit.entity.Message;
+import com.sprint.mission.discodeit.entity.ReadStatus;
+import com.sprint.mission.discodeit.entity.UserStatus;
+import com.sprint.mission.discodeit.repository.ChannelRepository;
+import com.sprint.mission.discodeit.repository.MessageRepository;
+import com.sprint.mission.discodeit.repository.ReadStatusRepository;
+import com.sprint.mission.discodeit.service.ChannelService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.time.Instant;
+import java.util.*;
+import java.util.stream.Collectors;
+
+@Service
+@RequiredArgsConstructor
+public class BasicChannelService implements ChannelService {
+    private final ChannelRepository channelRepository;
+    private final ReadStatusRepository readStatusRepository;
+    private final MessageRepository messageRepository;
+
+    @Override
+    public Channel createPublic(CreatePublicChannelDTO createPublicChannelDTO) {
+        return channelRepository.save(new Channel(createPublicChannelDTO));
+    }
+
+    @Override
+    public Channel createPrivate(CreatePrivateChannelDTO createPrivateChannelDTO) {
+        Channel channel = new Channel(createPrivateChannelDTO.channelType());
+
+        createPrivateChannelDTO.userIds()
+                .forEach(userId ->
+                        readStatusRepository.save(new ReadStatus(new CreateReadStatusDTO(userId,channel.getId()))));
+
+        return channelRepository.save(channel);
+    }
+
+    @Override
+    public FindChannelDTO find(UUID id) {
+        Channel channel = channelRepository.findById(id)
+                .orElseThrow(() -> new NullPointerException("Channel not found"));
+
+        Message message = messageRepository.findAll().stream()
+                .filter(m -> m.getChannel().equals(id))
+                .findFirst().orElseThrow(() -> new NullPointerException("Channel not found"));
+
+        if(channel.getType().equals(ChannelType.PRIVATE)){
+            return new FindChannelDTO(channel,message.getCreated(),message.getSender());
+        }
+
+        return new FindChannelDTO(channel,message.getCreated(),null);
```

Comment:
세번째 인자인 userId를 null로 넣어줘도 괜찮긴 하지만,  
FindChannelDTO 내에 createPublicChannelDto라는 정적 팩토리 메서드를 만들면 코드의 가독성이 조금 더 좋아질 것 같아요.  

```java
public record FindChannelDTO(
        Channel channel,
        Instant recentMessage,
        UUID userId
) {

    // 추가
    public static FindChannelDTO createPublicChannelDto(Channel channel, Instant recentMessage) {
        return new FindChannelDTO(
                channel,
                recentMessage,
                null
        );
    }
}
```

```suggestion
        return FindChannelDTO.createPublicChannelDto(channel,message.getCreated());
```